src/model/TexImage.cpp

```diff
@@ -32,6 +32,9 @@ auto TexImage::clone() -> Element* {
     img->width = this->width;
     img->height = this->height;
     img->text = this->text;
+    img->snappedBounds = this->snappedBounds;
+    img->sizeCalculated = this->sizeCalculated;
+    img->pdf = this->pdf;
```

Comment:
Does this need a `g_object_ref`? From some testing, it seems to work without crashing, but the PDF is ref counted and we're holding onto a reference here.

Reply:
Hmm, I can't tell. If I replace the last line by     

```c++
img->pdf = (PopplerDocument*)g_object_ref(this->pdf);
```
I do get crashes when destroying all the clones I made of a TeXImage.

<details>
<summary> Error log </summary>

```
Error: signal 11
[bt]: (0) build/src/xournalpp(+0x3fe30e) [0x55870ed7130e]
[bt]: (1) /lib/x86_64-linux-gnu/libc.so.6(+0x41950) [0x7f11cbe7a950]
[bt]: (2) /usr/lib/x86_64-linux-gnu/libcairo.so.2(cairo_transform+0x4) [0x7f11cc9c56a4]
[bt]: (3) /usr/lib/x86_64-linux-gnu/libpoppler-glib.so.8(+0x44905) [0x7f11cc507905]
[bt]: (4) /usr/lib/x86_64-linux-gnu/libpoppler.so.102(_ZN3GfxC2EP6PDFDocP9OutputDeviP4DictddPK12PDFRectangleS8_iPFbPvES9_P4XRef+0x1ff) [0x7f11cb1551df]
[bt]: (5) /usr/lib/x86_64-linux-gnu/libpoppler.so.102(_ZN4Page9createGfxEP9OutputDevddibbiiiibPFbPvES2_P4XRef+0x211) [0x7f11cb1af6d1]
[bt]: (6) /usr/lib/x86_64-linux-gnu/libpoppler.so.102(_ZN4Page12displaySliceEP9OutputDevddibbiiiibPFbPvES2_PFbP5AnnotS2_ES2_b+0x151) [0x7f11cb1b0291]
[bt]: (7) /usr/lib/x86_64-linux-gnu/libpoppler-glib.so.8(+0x2706c) [0x7f11cc4ea06c]
[bt]: (8) build/src/xournalpp(_ZN12DocumentView12drawTexImageEP6_cairoP8TexImage+0x238) [0x55870ed63ba4]
[bt]: (9) build/src/xournalpp(_ZNK12DocumentView11drawElementEP6_cairoP7Element+0x191) [0x55870ed63ee3]
[bt]: (10) build/src/xournalpp(_ZN12DocumentView13drawSelectionEP6_cairoP16ElementContainer+0x9f) [0x55870ed64281]
[bt]: (11) build/src/xournalpp(_ZN21EditSelectionContents5paintEP6_cairodddddd+0x298) [0x55870ec646b0]
[bt]: (12) build/src/xournalpp(_ZN13EditSelection5paintEP6_cairod+0x22c) [0x55870ec5e74e]
[bt]: (13) build/src/xournalpp(+0x3b2d06) [0x55870ed25d06]
[bt]: (14) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3b6204) [0x7f11ccf8a204]
[bt]: (15) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f11ccd565b5]
[bt]: (16) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1826bd) [0x7f11ccd566bd]
[bt]: (17) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2b2c91) [0x7f11cce86c91]
[bt]: (18) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3a68e9) [0x7f11ccf7a8e9]
[bt]: (19) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x187b71) [0x7f11ccd5bb71]
[bt]: (20) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18d6e9) [0x7f11ccd616e9]
[bt]: (21) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3a7711) [0x7f11ccf7b711]
[bt]: (22) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3b6204) [0x7f11ccf8a204]
[bt]: (23) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f11ccd565b5]
[bt]: (24) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1826bd) [0x7f11ccd566bd]
[bt]: (25) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2e2b8d) [0x7f11cceb6b8d]
[bt]: (26) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x187b71) [0x7f11ccd5bb71]
[bt]: (27) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18d6e9) [0x7f11ccd616e9]
[bt]: (28) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2e0ad5) [0x7f11cceb4ad5]
[bt]: (29) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3b6204) [0x7f11ccf8a204]
[bt]: (30) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f11ccd565b5]
[bt]: (31) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1826bd) [0x7f11ccd566bd]
[bt]: (32) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x132178) [0x7f11ccd06178]
[bt]: (33) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x187b71) [0x7f11ccd5bb71]
[bt]: (34) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18d6e9) [0x7f11ccd616e9]
[bt]: (35) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x134bb5) [0x7f11ccd08bb5]
[bt]: (36) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3b6204) [0x7f11ccf8a204]
[bt]: (37) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f11ccd565b5]
[bt]: (38) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x28f53c) [0x7f11cce6353c]
[bt]: (39) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x187b71) [0x7f11ccd5bb71]
[bt]: (40) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18d6e9) [0x7f11ccd616e9]
[bt]: (41) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x28f345) [0x7f11cce63345]
[bt]: (42) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3b6204) [0x7f11ccf8a204]
[bt]: (43) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f11ccd565b5]
[bt]: (44) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1826bd) [0x7f11ccd566bd]
[bt]: (45) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x132178) [0x7f11ccd06178]
[bt]: (46) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x187b71) [0x7f11ccd5bb71]
[bt]: (47) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18d6e9) [0x7f11ccd616e9]
[bt]: (48) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x134bb5) [0x7f11ccd08bb5]
[bt]: (49) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3b6204) [0x7f11ccf8a204]
[bt]: (50) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f11ccd565b5]
[bt]: (51) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1826bd) [0x7f11ccd566bd]
[bt]: (52) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x132178) [0x7f11ccd06178]
[bt]: (53) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x187b71) [0x7f11ccd5bb71]
[bt]: (54) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18d6e9) [0x7f11ccd616e9]
[bt]: (55) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x134bb5) [0x7f11ccd08bb5]
[bt]: (56) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3b6204) [0x7f11ccf8a204]
[bt]: (57) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f11ccd565b5]
[bt]: (58) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1826bd) [0x7f11ccd566bd]
[bt]: (59) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3b6204) [0x7f11ccf8a204]
[bt]: (60) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_container_propagate_draw+0x235) [0x7f11ccd565b5]
[bt]: (61) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1826bd) [0x7f11ccd566bd]
[bt]: (62) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3c4f1a) [0x7f11ccf98f1a]
[bt]: (63) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3b6204) [0x7f11ccf8a204]
[bt]: (64) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(+0x3bf9a0) [0x7f11ccf939a0]
[bt]: (65) /usr/lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_main_do_event+0x8c8) [0x7f11cce2fb18]
[bt]: (66) /usr/lib/x86_64-linux-gnu/libgdk-3.so.0(+0x3de89) [0x7f11ccb07e89]
[bt]: (67) /usr/lib/x86_64-linux-gnu/libgdk-3.so.0(+0x4f8a1) [0x7f11ccb198a1]
[bt]: (68) /usr/lib/x86_64-linux-gnu/libgdk-3.so.0(+0x514d1) [0x7f11ccb1b4d1]
[bt]: (69) /usr/lib/x86_64-linux-gnu/libgdk-3.so.0(+0x516c8) [0x7f11ccb1b6c8]
[bt]: (70) /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0(+0x14b56) [0x7f11cc6ebb56]
[bt]: (71) /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0(g_signal_emit_valist+0xf9f) [0x7f11cc704bbf]
[bt]: (72) /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0(g_signal_emit+0x93) [0x7f11cc704da3]
[bt]: (73) /usr/lib/x86_64-linux-gnu/libgdk-3.so.0(+0x4809e) [0x7f11ccb1209e]
[bt]: (74) /usr/lib/x86_64-linux-gnu/libgdk-3.so.0(+0x31fbd) [0x7f11ccafbfbd]
[bt]: (75) /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0(+0x53f98) [0x7f11cd3f3f98]
[bt]: (76) /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_dispatch+0x15f) [0x7f11cd3f33df]
[bt]: (77) /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0(+0x53788) [0x7f11cd3f3788]
[bt]: (78) /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_iteration+0x33) [0x7f11cd3f3853]
[bt]: (79) /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0(g_application_run+0x1cd) [0x7f11cc81854d]
[bt]: (80) build/src/xournalpp(_ZN11XournalMain3runEiPPc+0x5ef) [0x55870ebf48a2]
[bt]: (81) build/src/xournalpp(main+0x29) [0x55870ebb3302]
[bt]: (82) /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf2) [0x7f11cbe61cb2]
[bt]: (83) build/src/xournalpp(_start+0x2e) [0x55870ebb321e]
```
</details>